ipn/prefs.go

```diff
@@ -203,6 +208,14 @@ type Prefs struct {
 	Persist *persist.Persist `json:"Config"`
 }
 
+// AutoUpdatePrefs are the auto update settings for the node agent.
+type AutoUpdatePrefs struct {
+	// Enable specifies whether background auto-updates are enabled. When
+	// enabled, tailscaled will periodically check for available updates and
+	// apply them.
+	Enable bool
```

Comment:
I still think this is too coarse of a setting.

It's really like four things:

* check / notify
* auto-download the updates (and verify them) but don't install/restart.
* allow remote updates (c2n triggered) based on smart schedule, admin clicks
* do it locally

This lumps it all together into one bool and I'm afraid we're going to end up in a state state (even temporarily) where tailscale nodes are randomly rebooting on users at inopportune times and people will turn it off.

Reply:
Split the flag into separate `Check` and `Apply` flags. We can add a flag for accepting remote updates when that's actually implemented (and default it to "true" then).  For the initial Beta we're making auto-updates opt-in and checks/notifications opt-out.

For schedule, should we do a time window? A daily check time? Some other spec?
What would you want for your nodes?

When would users want to auto-download updates but not apply them?

Reply:
@bradfitz as discussed, we will do most of the update triggering via c2n messages from coordination server.
I'd still like these flags, to later hook them up to https://github.com/tailscale/tailscale/blob/9c4364e0b764e39084f207347b5cfd0dfba5a59d/ipn/ipnlocal/c2n.go#L110
The `--update-check` will let the coordination send update c2n messages to the client.
The `--update-apply` will gate whether client actually applies them.
WDYT?